### DISCUSSION TITLE
ModuleNotFoundError: No module named 'fc9c2cf28d98803df16d__mypyc'

### DISCUSSION THREAD

**Discussion by bastienexertier** [pyinstaller/pyinstaller#7999] [Help] [answered]
### Issue
Hello,
I am trying to package a simple script
```python
import lagom
```
I can succesfully produce the executable with
```sh
pyinstaller --onefile t.py
```
But when I try to run the executable I get this error:
```
Traceback (most recent call last):
  File "t.py", line 2, in <module>
  File "PyInstaller\loader\pyimod02_importers.py", line 391, in exec_module
  File "lagom\__init__.py", line 2, in <module>
ModuleNotFoundError: No module named 'fc9c2cf28d98803df16d__mypyc'
[3232] Failed to execute script 't' due to unhandled exception!
```
I think this is because `lagom` uses `mypycify` in the [setup.py](https://github.com/meadsteve/lagom/blob/master/setup.py) to compile the sources.
We can see that we can set the env var `LAGOM_SKIP_COMPILE=1` before running `pip install lagom` to disable mypy compilation but this doesn't fix the issue.

I'm not sure how to disable this compilation or how to include mypy compiled file with pyinstaller.
Thanks for your help.

### Full Repro

```sh
mkdir module_not_found_mypyc
cd module_not_found_mypyc
set LAGOM_SKIP_COMPILE=1
pip install lagom
pip install pyinstaller
echo "import lagom" > t.py
pyinstaller --onefile t.py
dist\t.exe
```

**Comment by bastienexertier** [comment]
Deleting every *.pyd in \Lib\site-packages\lagom seems to solve the issue, maybe my issue is more related to lagom/mypy instead of pyinstaller

**Comment by rokm** [comment, marked as answer]
PyInstaller cannot analyze imports in binary extensions, so if a module is imported only in binary extensions, we will fail to collect it. So such modules need to be manually specified as hidden imports.

In this particular case, you will need to add `--hiddenimport fc9c2cf28d98803df16d__mypyc` and `--collect-submodules lagom` to your PyInstaller command. The former will ensure that `fc9c2cf28d98803df16d__mypyc` is collected (as you can already see it is missing). Afterwards, you will see an error about missing `lagom` modules, which is caused by the same limitation; the missing modules are imported only in extensions, and are thus not automatically picked up. So `--collect-submodules` ensures that all modules (and extensions) from that package are picked up (note that the `fc9c2cf28d98803df16d__mypyc` is located outside of the package).

**Comment by bastienexertier** [comment]
Thank you it worked!
